src/Compilers/Core/Portable/CodeGen/LocalSlotManager.cs

```diff
@@ -64,6 +64,23 @@ public override bool Equals(object? obj)
         // maps local identities to locals.
         private Dictionary<ILocalSymbolInternal, LocalDefinition>? _localMap;
 
+        // The lowered tree might define the same local symbol
+        // in multiple sequences that are part of one expression, for example:
```

Comment:
>in multiple sequences that are part of one expression, for example: [](http://example.com/codeflow?start=11&length=67)

This doesn't sound right.  I think we should fix this, a local symbol should belong to exactly one scope in the bound tree. I assume this is a pre-existing condition, i.e. it is not introduced by this change. If so, I would prefer the fix to go into a separate PR.

Reply:
Sounds good, I will look into that.

Just to confirm: I think this "invalid" tree shape is introduced somewhere in lowering simply by reusing the same node twice, e.g., something like `_factory.Call(arguments: [node, node])`, and that `node` happens to contain a BoundSequence which then causes the local to be declared twice in the tree. So are you saying we should never do that (reuse the same node) and if we do it somewhere, it's a bug (hopefully a rare one)?

Reply:
In general, we can reuse bound nodes as long as that doesn't violate some invariants. For example, we reuse some primitive nodes: for constants, for local references (BoundLocal), etc. I consider "a local symbol used in a tree should belong to exactly one scope in the the tree" as one of the invariants that is violated in the specific scenario.